src/target/macos.rs

```diff
@@ -19,6 +26,56 @@ use std::{
 // For more information about ASLR, you can refer to https://en.wikipedia.org/wiki/Address_space_layout_randomization
 const _POSIX_SPAWN_DISABLE_ASLR: i32 = 0x0100;
 
+// Max number of characters to read from a thread name.
+const MAX_THREAD_NAME: usize = 100;
+
+struct OSXThread {
+    port: mach_port_name_t,
+    pthread_id: Option<usize>,
+    task_port: ipc_space_t,
+}
+
+impl Drop for OSXThread {
+    fn drop(&mut self) {
+        let result = unsafe { mach::mach_port::mach_port_deallocate(self.task_port, self.port) };
+        if result != kern_return::KERN_SUCCESS {
+            panic!("Failed to deallocate port!");
+        }
+    }
+}
+
+extern "C" {
+    // FIXME: Use libc  > 0.2.74 when available
```

Comment:
You mean this will be in the next release of the `libc` crate?

Reply:
Yes.